app/invocation/cache_requests_card.tsx

```diff
@@ -368,7 +371,62 @@ export default class CacheRequestsCardComponent extends React.Component<CacheReq
     ));
   }
 
+  private getCacheMetadata(scorecardResult: cache.ScoreCard.IResult) {
+    const digest = scorecardResult.digest;
+    const remoteInstanceName = this.props.model.getRemoteInstanceName();
+    rpc_service.service
+      .getCacheMetadata(
+        new cache.GetCacheMetadataRequest({
+          resourceName: {
+            digest: digest,
+            cacheType: scorecardResult.cacheType,
+            instanceName: remoteInstanceName,
+          },
+        })
+      )
+      .then((response) => {
+        this.setState(function (prevState, props) {
+          prevState.digestToCacheMetadata.set(digest.hash, response);
+          return {
+            digestToCacheMetadata: prevState.digestToCacheMetadata,
+          };
+        });
+      })
+      .catch((e) => {
+        console.log("Could not fetch metadata: " + BuildBuddyError.parse(e));
+
+        // Set an empty struct in the map, so FE doesn't keep trying to fetch an invalid result
+        this.setState(function (prevState, props) {
+          prevState.digestToCacheMetadata.set(digest.hash, null);
+          return {
+            digestToCacheMetadata: prevState.digestToCacheMetadata,
+          };
+        });
```

Comment:
nit: if you move this logic to just before you begin the fetch, that will also fix another issue that if you hover again while a request is in progress, you could potentially fire a duplicate request for the same digest.

also same comment as above re. setState() vs. forceUpdate()

Reply:
Nice - done